dev-docs/007-transforms-rfc.md

```diff
@@ -0,0 +1,228 @@
+   would be to define it's behavior as having no _pre-defined_ transform applied to the data. This
+   would allow the use of the newly defined _links_ structure with any stream by default without any
+   backwards compatibility issue.
+
+Below is the full list of behavior for the new _links_ data with regard to the stream coordinate.
+
+Streams with a coordinate:
+
+- _IDENTITY_ would have any _links_ entries applied to the stream
+- _VEHICLE_RELATIVE_ would apply _links_ if an entry is found. If an entry is not found, then the
+  default pose _/vehicle_pose_ would be used
+- _GEOGRAPHIC_ would not apply _links_ as it already defines a coordinate frame
+- _DYNAMIC_ calls a function which has access to the internal stream state. This would include any
+  _links_ defined so it may use that data as desired.
+
+Of particular note is that the _VEHICLE_RELATIVE_ behavior makes it now possible to support multiple
+vehicles.
+
+2. Adding the _links_ field to the state_update message.
+
+A new object would be added to the streamset specification.
+
+- The field name would be _links_
+- The keys of the _links_ object would represent the target stream the link affects
+  - The target could be a _primitive_ stream or a _pose_ stream
+- The value of a key would be an object that contains an entry _source_pose_ which is a string that
+  names a _pose_ stream
+
+3. Remove the timestamp on the Pose
+
+This is an artifact of an earlier verion of XVIZ. Today the streamset defines the timestamp, with
+the pose timestamp being a fallback. This fallback should be removed and the field from pose
+removed.
+
+4. Mixing of links between message types
+
+Links and poses create connections between streams that is expected to span the various state_update
+message types.
+
+A common distinction is made between static and dynamic frames. An example would be a static
+reference frame defined as `/world_base` which defines the origin. We could then have a dynamic
+reference frame `/vehicle_pose` that is a moving platform relative to `/world_base`. Next we can
+define a static reference frame for a lidar sensor relative to `/vehicle_pose`.
+
+This creates a chain that is static frame -> dynamic frame -> static frame.
+
+In this chain, only the dynamic frame will be updated but that update would cascade to any data that
+references the final static frame.
+
+In XVIZ a static entry would be sent in a _PERSISTENT_ state_update message. Dyanmic links and poses
+would come from a _COMPLETE_ or _INCREMENTAL_ message.
+
+5. Missing or invalid transform chains
+
+If there are holes in the transform chain then the stream should be marked with an error such that
+the viewing client can communicate the nature of the error to the users.
+
+## Message combinations
+
+State update messages have COMPLETE, INCREMENTAL and PERSISTENT types.
+
+### COMPLETE and INCREMENTAL
+
+These modes fundamentally deal with creating and adding data that is subject to the TIME_WINDOW
```

Comment:
I would define what this is in document.  I don't think we wanted this to apply to everyone right?

Reply:
You mean TIME_WINDOW.  I can expand that it is a config setting that limits the Window an application assumes data to be valid within, relative to a current timestamp.

This is a nuance to XVIZ really.  One could implement a XVIZ state storage that completely ignored TIME_WINDOW.  It's only relevant for the current spec.

I tried to call out specifically that links go in persistence as well as static transforms, etc.